v2/tools/generator/internal/codegen/pipeline/stage.go

```diff
@@ -253,3 +268,9 @@ func (stage *Stage) Postrequisites() []string {
 func (stage *Stage) Targets() []Target {
 	return stage.targets
 }
+
+// AddDiagnostic specifies a diagnostic generator for this stage.
+// The generator will be used if the generator is run with a --debug flag`
+func (stage *Stage) AddDiagnostic(diagnostic diagnosticAction) {
```

Comment:
Since `AddDiagnostic` is exported, should `DiagnosticAction` be too?

Reply:
Sure.